app/src/main/java/ch/logixisland/anuto/entity/enemy/Enemy.java

```diff
@@ -177,8 +179,14 @@ public float getSpeed() {
         return mEnemyProperties.getSpeed() * Math.max(mSpeedModifier, GameSettings.MIN_SPEED_MODIFIER);
     }
 
-    public void modifySpeed(float f) {
-        mSpeedModifier = mSpeedModifier * f;
+    public void modifySpeed(float f, Entity origin) {
+        if (origin instanceof Tower) {
+            Tower originTower = (Tower) origin;
+
+            if (!mEnemyProperties.getStrongAgainst().contains(originTower.getWeaponType())) {
```

Comment:
Why not doing this check in `GlueEffect`?

Reply:
You are right, since `modifySpeed` is only used by `GlueEffect`, I thought about that approach, too.

But I voted for the current solution, since the `getWeakAgainst`/`getStrongAgainst` check is here in `Enemy`-`damage`, as well.

And, well, who knows what cool new `Tower` the future will bring us :-)